src/multiselect_draggable.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * @license
+ * Copyright 2024 MIT
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Multiple selection draggable class.
+ */
+import * as Blockly from 'blockly/core';
+import {hasSelectedParent} from './global';
+
+
+/**
+ * A draggable object that adds the functionality for multiple blocks to
+ * be moved while someone is dragging it.
+ * It implements the IDraggable interface in Blockly v11.
+ */
+export class MultiselectDraggable {
+  /**
+   * A constructor for the MultiselectDraggable class
+   * @param {Blockly.Workspace} workspace The workspace the
+   * multiselectDraggable object is located in.
+   */
+  constructor(workspace) {
+    this.workspace = workspace;
+    this.id = Blockly.utils.idGenerator.genUid();
+    this.subDraggables = new Map();
+    this.topSubDraggables = [];
+    this.loc = new Blockly.utils.Coordinate(0, 0);
+    this.connectionDBList = [];
+  }
+
+  /**
+   * Clears everything in the subDraggables
+   * map of the MultiselectDraggable object.
+   */
+  clearAll() {
```

Comment:
I can see there are no private methods in this class, which is not a good idea. Private methods are suffixed with an underscore (`_`) (e.g. https://github.com/mit-cml/workspace-multiselect/blob/main/src/multiselect_controls.js#L235) You might actually want to make all the methods that don't belong to the interface private.